### PR TITLE
Add Parametric Soft Exponential Unit (PSEU) activation layer

### PR DESCRIPTION
**Pull request by SriRangaTarun**
**- What I did**
See above

**- How I did it**

**- How you can verify it**
I added unit tests

---
This pull request fixes #issue_number_here

### CODE REVIEW

**Comment by SriRangaTarun** [comment]
@gabrieldemarmiesse I have added the Parametric Soft Exponential Unit (PSEU) activation layer. Here is the [research paper on arXiv](https://arxiv.org/pdf/1602.01321.pdf). The build is passing now. Do you know someone who can review this ?

**Comment by gabrieldemarmiesse** [comment]
@SriRangaTarun We're currently focused on getting the design of this repo right to have a v1.0. So there might be delays to reviews PRs which add new features.

File: keras_contrib/layers/advanced_activations/pseu.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+from keras import initializers
```
Comment (RaphaelMeudec) [inline keras_contrib/layers/advanced_activations/pseu.py]: Usually, we like to have the imports sorted alphabetically, with first all the `import XXX` and then all the `from XX import YY`. See [this StackOverflow thread](https://stackoverflow.com/questions/20762662/whats-the-correct-way-to-sort-python-import-x-and-from-x-import-y-statement)

**Comment by RaphaelMeudec** [comment]
@SriRangaTarun Thanks for the contribution! I've left some comments. Also, as soon as https://github.com/keras-team/keras-contrib/pull/441 gets merged, you can rebase on it and add yourself as the code owner for this feature. Nice work!

File: keras_contrib/layers/advanced_activations/pseu.py (hunk 1 above)
Comment (SriRangaTarun) [reply keras_contrib/layers/advanced_activations/pseu.py]: Fixed

File: keras_contrib/layers/advanced_activations/pseu.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+                                      regularizer=self.regularizer,
+                                      constraint=self.constraint)
+        if self.trainable:
+            self.trainable_weights = [self.alphas]
+        if self.alpha_init is not None:
+            self.set_weights([self.alpha_init * np.ones(new_input_shape)])
+
+        self.build = True
+
+    def call(self, x, mask=None):
```
[the remark this answers is not in the document]
Comment (SriRangaTarun) [reply keras_contrib/layers/advanced_activations/pseu.py]: Removed. It has no impact.

File: keras_contrib/layers/advanced_activations/pseu.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        if self.alpha_init is not None and self.alpha_init < 0:
```
[the remark this answers is not in the document]
Comment (SriRangaTarun) [reply keras_contrib/layers/advanced_activations/pseu.py]: Removed.

File: keras_contrib/layers/__init__.py [hunk 4]
```diff
@@ -4,6 +4,7 @@
 from .advanced_activations.srelu import SReLU
 from .advanced_activations.swish import Swish
 from .advanced_activations.sinerelu import SineReLU
+from .advanced_activations.pseu import PSEU
```
[the remark this answers is not in the document]
Comment (SriRangaTarun) [reply keras_contrib/layers/__init__.py]: Fixed.

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec I have made all the changes you suggested and the CI build is still passing.

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse PR looks good to me. You might want to add @SriRangaTarun in your current codeowner PR?

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec and @gabrieldemarmiesse Everything is working fine. I have added important parameters and handled all possible input cases. But, since [the codeowners PR](https://github.com/keras-team/keras-contrib/pull/441) has been merged, it is failing the codeowners test.

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec and @gabrieldemarmiesse Today I tried to add initializer and trainable options, instead of only having a constant alpha initializer. But, I removed them because they cause numerical instability in the function (it is a piecewise function in alpha, so it becomes unstable when the negative formula is used for positive alpha weights and the other way round). A custom initializer or trainability can cause the signs of the weights to mismatch the function and result in NaNs.  So, I am sticking to constant initial alpha weights which can not be trained.

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec and @gabrieldemarmiesse Everything seems to work fine. But, ever since [the codeowners PR](https://github.com/keras-team/keras-contrib/pull/441) has been merged, it is failing the codeowners test.

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec and @gabrieldemarmiesse Today I tried to add initializer and trainable options, instead of only having a constant alpha initializer. But, I removed them because they cause numerical instability in the function (it is a piecewise function in alpha, so it becomes unstable when the negative formula is used for positive alpha weights and the other way round). A custom initializer or trainability can cause the signs of the weights to mismatch the function and result in NaNs. So, I am sticking to constant initial alpha weights which can not be trained.

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec and @gabrieldemarmiesse Today I tried to add initializer and trainable options, instead of only having a constant alpha initializer. But, I removed them because they cause numerical instability in the function (it is a piecewise function in alpha, so it becomes unstable when the negative formula is used for positive alpha weights and the other way round). A custom initializer or trainability can cause the signs of the weights to mismatch the function and result in NaNs. So, I am sticking to constant initial alpha weights which can not be trained.

**Comment by SriRangaTarun** [comment]
Now the build is passing after adding pseu.py to CODEOWNERS.

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec and @gabrieldemarmiesse Today I tried to add initializer and trainable options, instead of only having a constant alpha initializer. But, I removed them and reverted to the previous version because they cause numerical instability in the function (it is a piecewise function in alpha, so it becomes unstable when the negative formula is used for positive alpha weights and the other way round). A custom initializer or trainability can cause the signs of the weights to mismatch the function and result in NaNs. So, I am sticking to constant initial alpha weights which can not be trained.

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec Do you suggest any other changes ?

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec Do you suggest any other changes ? The most recent build failed because the instancenormalization test suddenly failed for some reason.

**Comment by RaphaelMeudec** [comment]
@SriRangaTarun All good, nothing to add IMO!

@gabrieldemarmiesse Failing test doesn't seem linked to this PR, code looks good to me!

**Comment by SriRangaTarun** [comment]
Somehow, the build is passing now.

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse I think we can merge this!

**Comment by gabrieldemarmiesse** [comment]
We'll merge this after I merge the tf.keras PR. It's getting quite painful to resolve conflicts every two days.

**Comment by gabrieldemarmiesse** [comment]
@SriRangaTarun Please merge master into your branches to make sure that it's compatible with tf.keras

**Comment by SriRangaTarun** [comment]
@RaphaelMeudec and @gabrieldemarmiesse I am closing this PR and have created an [alias here](https://github.com/keras-team/keras-contrib/pull/459). There, I have merged master into my branch.
Any other changes ?

**Pull request closed without merging**